### ISSUE TITLE
Update SOFA Version

### ISSUE THREAD

**Issue by xlxs4** [sisl/SOFA.jl#6]
Any plans on updating to a more recent SOFA version?

**Comment by duncaneddy** [comment]
I'll be honest, not in the immediate future. My life is a bit hectic currently, but I should be able to look at it sometime in January.

In the mean time, I'd also be happy to accept a pull request updating to the latest version.

**Comment by duncaneddy** [comment]
Looking at the package again, as written would be a significant manual update. 

This package should be refactored to automatically generate the Julia bindings from the underlying C source using something like https://github.com/analytech-solutions/CBinding.jl so that a SOFA version update is as easy as dropping in the new source files.

I'll put that refactor on my list, but no promises on timeline.

**Comment by icweaver** [comment]
Hi Abhro, I'm not sure if the PR you linked would close things here since it is only using the 2019 version. It sounds like the 2023 version might require some special treatment, but I am not familiar with the history here yet

**Comment by icweaver** [comment]
From a historical / code reference standpoint, would we want to prepare releases for any of the other sofa lib versions between 2019 and 2023 as well? cc: @barrettp

A brief claude search produces the following summary of changes that seem to have occurred:

---

## SOFA C library releases between 2019-07-22 and 2023-10-11

| SOFA date     | What changed | New routines |
|---------------|---|---|
| 2019-07-22    | (starting point — what `SOFA_jll` 2019.7.22+0 ships) | — |
| 2020-07-21    | Bug fixes only | none |
| 2021-01-25    | Bug and accuracy fixes | none |
| 2021-01-25_a  | Small bug-fix patch on top of 2021-01-25 | none |
| 2021-05-12    | Bug fixes + 3 new routines | `iauAtcc13`, `iauAtccq`, `iauMoon98` |
| 2023-10-11    | Bug fixes only (STARPV/PVSTAR machine-precision tweaks, `IYV` bumped to 2023, comment corrections in A2TF/ATOIQ/C2T06A/FK45Z/FK54Z/HFK5Z/LTPB, `t_atccq` test fix) | none |

The only release with new API surface in this span is **2021-05-12**, which adds three routines. All other intermediate releases are bug fixes only, with no signature changes.

Sources:
- [ERFA README.rst — "Differences from SOFA"](https://github.com/liberfa/erfa/blob/master/README.rst)
- [ERFA commit b2039ecd "Updating to SOFA 20210512"](https://github.com/liberfa/erfa/commit/b2039ecd)
- [SOFA changes page (2023-10-11 release notes)](https://www.iausofa.org/changes)

---

**Comment by barrettp** [comment]
I'm not sure that I understand your question given that SOFA.jl will become the Astrometry/SOFA submodule, which contains the entire SOFA functional API once it is transferred and therefore be updated to the latest version of SOFA.

**Comment by icweaver** [comment]
Right, that sounds good. My questions was more about the tagged wrapper releases we have here now, and if we wanted to include other ones just as a point of reference, separate from the Astrometry pure Julia version that will be included in a future breaking release. I don't have any allegiance either way, so just wanted to ask others what they would like

**Comment by barrettp** [comment]
It is not something that I am concerned about. However, if someone has the will and time, then they welcome to do it.

**Comment by icweaver** [comment]
Sounds good. I might have one of those, will try taking a look in a bit

**Comment by icweaver** [comment]
Tracking here: https://github.com/JuliaAstro/SOFA.jl/pull/28

**Comment by icweaver** [comment]
Howdy, we have releases up to the current SOFA Release 19 (v2023.10.11) now. Please feel free to re-open if there are any issues

https://github.com/JuliaAstro/SOFA.jl/releases
